mg.js

```diff
@@ -1,72 +1,84 @@
-const backImg = "img/back.jpeg";
-const frontImg = [
-  "img/front.jpg",
-  "img/front1.jpeg",
-  "img/front2.jpeg",
-  "img/front3.jpeg",
-  "img/front4.avif",
-  "img/front5.jpeg",
-];
-const doubledFrontImgs = [...frontImg, ...frontImg];
-let flipCount = 0;
-let startTime = null;
-let timerInterval = null;
-let flippedCards = [];
-const flipBackDelay = 2000;
 
-function shuffleArray(array) {
-  for (let i = array.length - 1; i > 0; i--) {
-    const j = Math.floor(Math.random() * (i + 1));
-    [array[i], array[j]] = [array[j], array[i]];
-  }
-  return array;
-}
+document.addEventListener("DOMContentLoaded", function () {
+  let frontImg = [];
+  fetch(
+    "https://raw.githubusercontent.com/SpringrollBrothers/SpringrollBrothers.github.io/main/memory.json"
+  )
+    .then((response) => {
+      if (!response.ok) {
+        throw new Error("Could not get your data");
+      }
+      return response.json();
+    })
+    .then((data) => {
```

Comment:
Placing all of your app's code in the `then` callback makes it challenging to follow. It can easily lead to the [callback hell](https://medium.com/@raihan_tazdid/callback-hell-in-javascript-all-you-need-to-know-296f7f5d3c1). So you should move the `suffleArray` and `createCardButton` functions out of the `then` callback and create functions to initialise your app. The code would then look similar to the following:

```js
function awesomeFunction(data) {
  // Do something with data tinitialiseze the app
}

fetch('...')
  .then((response) => {
    ...
  })
  .then((data) => {
    awesomeFunction(data);
  })
  .catch(...);

```

Reply:
Dear mentor, i have tried to fix it, i dont know if you could have a review before i proceed to merge ? or should i just go ahead to merge? :)